tests/install/test.py

```diff
@@ -74,31 +75,47 @@ class TestNested:
                 ),
                 "bios": (),
             }[firmware],
-            vdis=[dict(name="vm1 system disk", size="100GiB", device="xvda", userdevice="0")],
+            vdis=([dict(name="vm1 system disk", size="100GiB", device="xvda", userdevice="0")]
+                  + ([dict(name="vm1 system disk mirror", size="100GiB", device="xvdb", userdevice="1")]
+                     if system_disk_config == "raid1" else [])
+                  ),
             cd_vbd=dict(device="xvdd", userdevice="3"),
             vifs=[dict(index=0, network_name=NETWORKS["MGMT"])],
         ))
     @pytest.mark.answerfile(
-        lambda system_disks_names, local_sr, package_source, iso_version: AnswerFile("INSTALL")
+        lambda system_disks_names, local_sr, package_source, system_disk_config, iso_version: AnswerFile("INSTALL")
         .top_setattr({} if local_sr == "nosr" else {"sr-type": local_sr})
         .top_append(
             {"TAG": "source", "type": "local"} if package_source == "iso"
             else {"TAG": "source", "type": "url",
                   "CONTENTS": ISO_IMAGES[iso_version]['net-url']} if package_source == "net"
             else ValueError(f"package_source {package_source!r}"),
+
+            {"TAG": "raid", "device": "md127",
+             "CONTENTS": [
+                 {"TAG": "disk", "CONTENTS": diskname} for diskname in system_disks_names
+             ]} if system_disk_config == "raid1"
+            else None if system_disk_config == "disk"
+            else ValueError(f"system_disk_config {system_disk_config!r}"),
+
             {"TAG": "admin-interface", "name": "eth0", "proto": "dhcp"},
             {"TAG": "primary-disk",
              "guest-storage": "no" if local_sr == "nosr" else "yes",
-             "CONTENTS": system_disks_names[0]},
+             "CONTENTS": ("md127" if system_disk_config == "raid1"
```

Comment:
I'm not a fan of nesting conditional expressions. I think I would prefer something like:

~~~python
dict(raid1='md127', disk=system_disks_names[0])[system_disk_config]
~~~

Reply:
I used those in a few places (I do like getting in a single short line what would otherwise take 4), but in this particular place we'll want to be consistent and not mix the 2 idioms, and I'm not sure how readable that would get with the larger expressions.
OTOH, getting real `ValueError` raised instead of having to add them in the type hints has a lot of appeal... this needs to be looked at.

Reply:
why didn't you raise the `ValueError` in the first place if that's an acceptable option?

I think that

```python
def raise_value_error(v) -> SimpleAnswerfileDict:
    raise ValueError(v)

{"TAG": "raid", "device": "md127",
  "CONTENTS": [
     {"TAG": "disk", "CONTENTS": diskname} for diskname in system_disks_names
 ],
} if system_disk_config == "raid1"
else None if system_disk_config == "disk"
else raise_value_error(f"system_disk_config {system_disk_config!r}")
```

should have worked (not tasted). Am I missing something?